ci/it/testcases/utils.go

```diff
@@ -34,19 +36,53 @@ type Containers struct {
 	ClickHouse    *testcontainers.Container
 }
 
-func (c *Containers) Cleanup(ctx context.Context) {
-	if c.Elasticsearch != nil {
-		(*c.Elasticsearch).Terminate(ctx)
+func printContainerLogs(ctx context.Context, container *testcontainers.Container, name string) {
+	if container == nil {
+		return
 	}
-	if c.Quesma != nil {
-		(*c.Quesma).Terminate(ctx)
+
+	reader, err := (*container).Logs(ctx)
+	if err != nil {
+		log.Printf("Failed to get logs for container '%s': %v", name, err)
+		return
+	}
+
+	output, err := io.ReadAll(reader)
+	if err != nil {
+		log.Printf("Failed to read logs for container '%s': %v", name, err)
+		return
+	}
+
+	log.Printf("Logs for container '%s':", name)
+	for _, line := range strings.Split(string(output), "\n") {
+		log.Printf("Container '%s': %s", name, line)
+	}
+}
+
+func terminateContainer(ctx context.Context, container *testcontainers.Container, name string) {
+	if container == nil {
+		log.Printf("Container '%s' is nil", name)
+		return
 	}
-	if c.Kibana != nil {
-		(*c.Kibana).Terminate(ctx)
+
+	err := (*container).Terminate(ctx)
+	if err != nil {
+		log.Printf("Failed to terminate container '%s': %v", name, err)
 	}
-	if c.ClickHouse != nil {
-		(*c.ClickHouse).Terminate(ctx)
+}
+
+func (c *Containers) Cleanup(ctx context.Context, t *testing.T) {
+	if t.Failed() {
+		printContainerLogs(ctx, c.Elasticsearch, "Elasticsearch")
+		printContainerLogs(ctx, c.Quesma, "Quesma")
+		printContainerLogs(ctx, c.Kibana, "Kibana")
+		printContainerLogs(ctx, c.ClickHouse, "ClickHouse")
```

Comment:
So far I only ever need Quesma logs, wouldn't it be too noisy? Maybe only print others if there is issue with container?

Reply:
We saw a couple IT fails where ClickHouse had problems. Similarly (maybe we haven't yet seen it in ITs) sometimes Elasticsearch has problems which causes problems with Quesma.

Reply:
Ok.